osu.Game/Screens/Play/Player.cs

```diff
@@ -213,6 +213,12 @@ private void load(OsuConfigManager config, OsuGameBase game, CancellationToken c
             if (playableBeatmap == null)
                 return;
 
+            if (gameplayMods.Any(m => ruleset.AllMods.All(rulesetMod => m.GetType() != rulesetMod.GetType())))
```

Comment:
Even if this is only run once during gameplay initialisation, it's insanely inefficient. Should be trivial to rewrite in a more efficient way.

Reply:
I tend to use LINQ whenever it's a once-off operation, keeps things short.